cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+		additionalEnvironment = append(additionalEnvironment, "CF_HOME="+config.CfHome)
+	}
+
+	if len(config.CfPluginHome) > 0 {
+		additionalEnvironment = append(additionalEnvironment, "CF_PLUGIN_HOME="+config.CfPluginHome)
+	}
+
+	log.Entry().Infof("Using additional environment variables: %s", additionalEnvironment)
+
+	// TODO set HOME to config.DockerWorkspace
+	command.SetEnv(additionalEnvironment)
+
+	err = _cfLogin(command, cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.APIEndpoint,
+		CfOrg:         config.Org,
+		CfSpace:       config.Space,
+		Username:      config.Username,
+		Password:      config.Password,
+		CfAPIOpts:     splitAtWhitespace(config.APIParameters),
+		CfLoginOpts:   splitAtWhitespace(config.LoginParameters),
+	})
+
+	if err == nil {
+		loginPerformed = true
+	}
+
+	if err == nil {
+		err = command.RunExecutable("cf", []string{"plugins"}...)
+		if err != nil {
+			log.Entry().WithError(err).Errorf("Command '%s' failed.", []string{"plugins"})
+		}
+	}
+
+	if err == nil {
+		err = command.RunExecutable("cf", cfDeployParams...)
+		if err != nil {
+			log.Entry().WithError(err).Errorf("Command '%s' failed.", cfDeployParams)
+		}
+	}
+
+	if err == nil && postDeployAction != nil {
+		err = postDeployAction(command)
+	}
+
+	if loginPerformed {
+
+		logoutErr := _cfLogout(command)
+
+		if logoutErr != nil {
+			log.Entry().WithError(logoutErr).Errorf("Cannot perform cf logout")
+			if err == nil {
+				err = logoutErr
+			}
+		}
+	}
+
+	if err != nil || GeneralConfig.Verbose {
+		e := handleCfCliLog(cfLogFile)
+		if e != nil {
+			log.Entry().WithError(err).Errorf("Error reading cf log file '%s'.", cfLogFile)
+		}
+	}
+
+	return err
+}
+
+func findMtar() (string, error) {
+
+	const pattern = "**/*.mtar"
+
+	mtars, err := fileUtils.Glob(pattern)
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(mtars) == 0 {
+		return "", fmt.Errorf("No mtar file matching pattern '%s' found", pattern)
+	}
+
+	if len(mtars) > 1 {
+		sMtars := []string{}
+		for _, mtar := range mtars {
+			sMtars = append(sMtars, mtar)
+		}
+		return "", fmt.Errorf("Found multiple mtar files matching pattern '%s' (%s), please specify file via mtaPath parameter 'mtarPath'", pattern, strings.Join(sMtars, ","))
+	}
+
+	return mtars[0], nil
+}
+
+func handleCfCliLog(logFile string) error {
+
+	fExists, err := fileUtils.FileExists(logFile)
+
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Info("### START OF CF CLI TRACE OUTPUT ###")
+
+	if fExists {
+
+		f, err := os.Open(logFile)
+
+		if err != nil {
+			return err
+		}
+
+		bReader := bufio.NewReader(f)
+
+		var done bool
+
+		for {
+			line, err := bReader.ReadString('\n')
+			if err == io.EOF {
+
+				// maybe inappropriate to log as info. Maybe the line from the
+				// log indicates an error, but that is something like a project
+				// standard.
+				done = true
+			} else if err != nil {
+
+				break
+			}
+
+			log.Entry().Info(strings.TrimSuffix(line, "\n"))
+
+			if done {
+				break
+			}
```

Comment:
Is io.EOF returned even when line still contains valid content that should be logged?

Reply:
What is the context for that question? In case `io.EOF` is returned and the line contains some content the content is logged nevertheless. We `break` _after_ logging that last line. From that point of view everything should be fine. Maybe we have some kind of misunderstanding ...

Reply:
I wrongly assumed that `line` and `err` are mutually exclusive. (That is an assumption which I hope you can sympathize with. :-D) But I have run this in a test, and indeed the last line is returned in `line` while `err` is `io.EOF`at the same time.

That being said, the code could be simplified a lot:
```golang
bReader := bufio.NewReader(f)
	for {
		line, err := bReader.ReadString('\n')
		if err == nil || err == io.EOF {
			log.Entry().Info(strings.TrimSuffix(line, "\n"))
		}
		if err != nil {
			break
		}
	}
```

Reply:
yes, thats much simpler --> adjusted accordingly